src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,40 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+	
+    public final String value;
+    private boolean isPrivate;
+    
+    public Contact(String contact, boolean isPrivate, String messageConstraints, String validationRegex) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        contact = contact.trim();
+        if (!isValidContact(contact, validationRegex)) {
+            throw new IllegalValueException(messageConstraints);
+        }
+        this.value = contact;
+    }
+    
+    public static boolean isValidContact(String test, String validationRegex) {
+        return test.matches(validationRegex);
+    }
+    
+    public String toString() {
+        return value;
+    }
+
+    public boolean equals(Object other) {
```

Comment:
This implementation will allow you to compare Email, Address and Phone objects. Although syntactically correct, it is not semantically accurate. You should implement this in each children class separately.